src/parse.rs

```diff
@@ -71,6 +71,31 @@ where
     }
 }
 
+/// Parse an expected single `T` Response with `F` and store it in `into`.
+/// Responses other than `T` go into the `unsolicited` channel.
+///
+/// If zero or more than one `T` is found then into will be `None`
+pub(crate) fn parse_until_done<'input, T, F>(
+    input: &'input [u8],
+    into: &mut Option<T>,
+    unsolicited: &mut mpsc::Sender<UnsolicitedResponse>,
+    map: F,
+) -> Result<()>
+where
+    F: FnMut(Response<'input>) -> Result<MapOrNot<'input, T>>,
+{
+    let mut temp_output = Vec::<T>::new();
```

Comment:
I think we _should_ be able to to have `parse_many_into` accept an `&mut Option` as `IT` directly. At least I thought `Option` implemented `Extend` as well.

Reply:
Huh, interesting. I could have sworn... Oh well, fair enough, let's do it this way then.